services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersSpec.groovy

```diff
@@ -209,6 +210,66 @@ on a #switchType switch"() {
         "non-Centec" | getNonCentecSwitches()
     }
 
+    @Unroll
+    def "Source/destination switches have meters only in flow ingress rules and intermediate switches don't have \
+meters in flow rules at all"() {
+        assumeTrue("Unable to find required switches in topology", switches.size() > 1)
+
+        given: "Two active not neighboring #switchType switches"
+        def allLinks = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allLinks.every { link -> !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId) }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        when: "Create a flow between these switches"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        then: "The source and destination switches have only one meter in the flow's ingress rule"
+        def srcSwFlowMeters = northbound.getAllMeters(srcSwitch.dpId).meterEntries.findAll {
+            it.meterId > MAX_SYSTEM_RULE_METER_ID
+        }
+        def dstSwFlowMeters = northbound.getAllMeters(dstSwitch.dpId).meterEntries.findAll {
+            it.meterId > MAX_SYSTEM_RULE_METER_ID
+        }
+
+        srcSwFlowMeters.size() == 1
+        dstSwFlowMeters.size() == 1
+
+        def srcSwitchRules = northbound.getSwitchRules(srcSwitch.dpId).flowEntries
+        def dstSwitchRules = northbound.getSwitchRules(dstSwitch.dpId).flowEntries
+        def srcSwFlowIngressRule = filterRules(srcSwitchRules, flow.source.portNumber, flow.source.vlanId, null)[0]
+        def dstSwFlowIngressRule = filterRules(dstSwitchRules, flow.destination.portNumber, flow.destination.vlanId,
+                null)[0]
+
+        srcSwFlowMeters[0].meterId == srcSwFlowIngressRule.instructions.goToMeter
+        dstSwFlowMeters[0].meterId == dstSwFlowIngressRule.instructions.goToMeter
+
+        and: "The source and destination switches have no meters in the flow's egress rule"
+        def srcSwFlowEgressRule = filterRules(srcSwitchRules, null, null, flow.source.portNumber)[0]
```

Comment:
I like the idea, but I have doubts whether it's safe to pick `[0]` value and ignore any other potential results returned by `filterRules`

Reply:
On the other hand, I think it's fine to assume that there's only one rule

Reply:
So there will be only one result because we create only one flow.